src/main/java/sorald/processor/SerialVersionUidProcessor.java

```diff
@@ -0,0 +1,191 @@
+package sorald.processor;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import sorald.annotations.IncompleteProcessor;
+import sorald.annotations.ProcessorAnnotation;
+import spoon.reflect.code.CtExpression;
+import spoon.reflect.declaration.CtClass;
+import spoon.reflect.declaration.CtElement;
+import spoon.reflect.declaration.CtField;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.declaration.ModifierKind;
+import spoon.reflect.reference.CtFieldReference;
+import spoon.reflect.reference.CtTypeReference;
+
+@IncompleteProcessor(
+        description =
+                "This processor ignores two corner cases: If the class has already a serialVersionUID with a non long type and if the class is not directly implementing serializable e.g. a upper class implements serializable.")
+@ProcessorAnnotation(
+        key = 2057,
+        description =
+                "Every class implementing Serializable should declare a static final serialVersionUID.")
+public class SerialVersionUidProcessor extends SoraldAbstractProcessor<CtClass<?>> {
+
+    private static final String DEFAULT_ID_VALUE = "1L";
+    private static final String SERIAL_VERSION_UID = "serialVersionUID";
+    private final Set<ModifierKind> modifiers = Set.of(ModifierKind.STATIC, ModifierKind.FINAL);
+
+    @Override
+    protected boolean canRepairInternal(CtClass<?> candidate) {
+        Set<CtTypeReference<?>> superInterfaces = candidate.getSuperInterfaces();
+        if (isException(candidate)) {
+            return false;
+        }
+        CtTypeReference<?> serializable =
+                candidate.getFactory().createCtTypeReference(Serializable.class);
+        if (superInterfaces.contains(serializable)) {
+
+            CtFieldReference<?> fieldReference = getSerialVersionUIDField(candidate);
+
+            if (fieldReference != null) {
+                CtField<?> field = fieldReference.getDeclaration();
+                if (!isLongType(field)) {
+                    // Here is the special case where the field exisits but it's type is wrong.
+                    // TODO: handle this case after discussion.
+                    return false;
+                }
+                return !hasCorrectModifier(field);
+            } else {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Checks if a given class ia a expection for having an ID. SonarQube states 4 expections
+     *
+     * <ul>
+     *   <li>Swing and AWT classes (GUI classes)
+     *   <li>abstract classes
+     *   <li>Throwable and its subclasses (Exceptions and Errors)
+     *   <li>classes marked with @SuppressWarnings("serial")
```

Comment:
Mention that this is noted as an exception in the docs, but isn't actually implemented as such.